app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/search/SearchViewModel.kt

```diff
@@ -726,9 +727,14 @@ data class SearchState(
          */
         @Parcelize
         data class Content(
-            val displayItems: List<DisplayItem>,
+            private val displayItems: List<DisplayItem>,
         ) : ViewState() {
             override val hasVaultFilter: Boolean get() = true
+
+            val displayItemsSorted: List<DisplayItem>
+                get() = displayItems.sortedWith { item1, item2 ->
+                    CompareStringSpecialCharWithPrecedence.compare(item1.title, item2.title)
```

Comment:
These seems like an expensive call to be done in the UI many times, can we do this sorting before we construct the `Content` so that it only runs the one time.

Reply:
I may be misunderstanding what `runs the one time` means in this case since the sorting would need to take place each time the list updates, which is triggered via the search term. I can move the the call to the sort on the creation of the new lists in the `SearchTypeDataExtensions.kt`. Just as a note even with it now happing in the calls through the action channel it is still technically happing on `Dispacther.MAIN` I am not seeing any rise in recomposition either. either way happy to take logic call sight out of the Composable